src/ja/server/database/sql/mock_database.py

```diff
@@ -24,93 +19,15 @@ def __init__(self) -> None:
-                job_entry.statistics.running_time = \
-                    (datetime.now() - job_entry.statistics.time_started).seconds - job_entry.statistics.paused_time
-            if old_job != job:
-                self.session.delete(old_job)
-                self.session.flush()
-                job_entry.job = deepcopy(job)
-        self.session.commit()
-
-    def get_jobs_on_machine(self, machine: WorkMachine) -> Optional[List[Job]]:
-        jobs: Optional[List[Job]] = self.session.query(Job).join(DatabaseJobEntry). \
-            join(WorkMachine, WorkMachine.uid == machine.uid).all()
-        return jobs
-
-    def assign_job_machine(self, job: Job, machine: WorkMachine) -> None:
-        job_entry = self.find_job_by_id(job.uid)
-        job_entry.assigned_machine = machine
-        self.session.commit()
+        def stat_call(j: Job) -> None:
+            pass
 
-    def update_work_machine(self, machine: WorkMachine) -> None:
-        work_machine = self.session.query(WorkMachine).filter(WorkMachine.uid == machine.uid).first()
-        if work_machine is None:
-            self.session.add(machine)
-        else:
-            self.session.delete(work_machine)
-            self.session.flush()
-            self.session.add(machine)
-        self.session.commit()
+        def scheduler_call(d: ServerDatabase) -> None:
+            pass
 
-    def get_work_machines(self) -> Optional[List[WorkMachine]]:
-        work_machines: Optional[List[WorkMachine]] = self.session.query(WorkMachine).all()
-        return work_machines
+        self.set_scheduler_callback(scheduler_call)
+        self.set_job_status_callback(stat_call)
+        self.in_scheduler_callback = False
 
-    def get_current_schedule(self) -> Optional[ServerDatabase.JobDistribution]:
-        jobs: Optional[List[DatabaseJobEntry]] = self.session.query(DatabaseJobEntry).join(Job) \
-            .filter((Job.status == JobStatus.RUNNING) | (Job.status == JobStatus.NEW) | (
-                Job.status == JobStatus.PAUSED) | (Job.status == JobStatus.QUEUED)).all()
-        return jobs
-
-    def query_jobs(self, since: datetime, user_id: Optional[int], work_machine: Optional[WorkMachine]) \
-            -> List[DatabaseJobEntry]:
-        jobs_query = self.session.query(DatabaseJobEntry)
-        if work_machine is not None:
-            jobs_query = jobs_query.join(WorkMachine).filter(WorkMachine.uid == work_machine.uid)
-        if user_id != -1:
-            jobs_query = jobs_query.join(Job).filter(Job.owner_id == user_id)
-        jobs: List[DatabaseJobEntry] = jobs_query.all()
-        if since is None:
-            return jobs
-        return [job for job in jobs if job.statistics.time_added >= since]
-
-    def set_scheduler_callback(self, callback: ServerDatabase.RescheduleCallback) -> None:
-        pass
-
-    def set_job_status_callback(self, callback: ServerDatabase.JobStatusCallback) -> None:
-        pass
+    def __del__(self) -> None:
```

Comment:
Is this needed? I thought Python will automatically call the parent's destructor.

Reply:
I have no idea, but if that what are you telling me is true then it is not really needed.